sbc_compassion/models/sponsorship_correspondence.py

```diff
@@ -343,3 +336,9 @@ def _get_read_url(self):
 
     def _get_uuid(self):
         return str(uuid.uuid4())
+
+    @api.model
+    def _needaction_domain_get(self):
+        domain = ['&', ('direction', '=', 'Beneficiary To Supporter'),
+                  ('state', '=', 'Published to Global Partner')]
```

Comment:
You can move this method in onramp_compassion module and add last_read to the domain.
Note : '&' is not necessary, it is the default operator for multiple conditions.